tests/models/bark/test_modeling_bark.py

```diff
@@ -0,0 +1,2486 @@
+            if model.get_output_embeddings() is None:
+                continue
+
+            # Check that resizing the token embeddings with a larger vocab size increases the model's vocab size
+            model_vocab_size = config.vocab_size
+            model.resize_token_embeddings(model_vocab_size + 10)
+            self.assertEqual(model.config.vocab_size, model_vocab_size + 10)
+            output_embeds_list = model.get_output_embeddings()
+
+            for output_embeds in output_embeds_list:
+                self.assertEqual(output_embeds.weight.shape[0], model_vocab_size + 10)
+
+                # Check bias if present
+                if output_embeds.bias is not None:
+                    self.assertEqual(output_embeds.bias.shape[0], model_vocab_size + 10)
+
+            # Check that the model can still do a forward pass successfully (every parameter should be resized)
+            model(**self._prepare_for_class(inputs_dict, model_class))
+
+            # Check that resizing the token embeddings with a smaller vocab size decreases the model's vocab size
+            model.resize_token_embeddings(model_vocab_size - 15)
+            self.assertEqual(model.config.vocab_size, model_vocab_size - 15)
+            # Check that it actually resizes the embeddings matrix
+            output_embeds_list = model.get_output_embeddings()
+
+            for output_embeds in output_embeds_list:
+                self.assertEqual(output_embeds.weight.shape[0], model_vocab_size - 15)
+                # Check bias if present
+                if output_embeds.bias is not None:
+                    self.assertEqual(output_embeds.bias.shape[0], model_vocab_size - 15)
+
+            # Check that the model can still do a forward pass successfully (every parameter should be resized)
+            # Input ids should be clamped to the maximum size of the vocabulary
+            inputs_dict["input_ids"].clamp_(max=model_vocab_size - 15 - 1)
+
+            # Check that the model can still do a forward pass successfully (every parameter should be resized)
+            model(**self._prepare_for_class(inputs_dict, model_class))
+
+
+@require_torch
+class BarkModelTest(unittest.TestCase):
+    @cached_property
+    def model(self):
+        return BarkModel.from_pretrained("ylacombe/bark-large").to(torch_device)
+
+    @cached_property
+    def processor(self):
+        return BarkProcessor(repo_id="ylacombe/bark-large", subfolder="speaker_embeddings")
+
+    @cached_property
+    def inputs(self):
+        input_ids, history_prompt = self.processor(
+            "In the light of the moon, a little egg lay on a leaf", voice_preset="en_speaker_6"
+        )
+
+        input_ids = {key: input_ids[key].to(torch_device) for key in input_ids}
+
+        return input_ids, history_prompt
+
+    @slow
+    def test_generate_text_semantic(self):
+        input_ids, history_prompt = self.inputs
+
+        expected_output_ids = [
```

Comment:
As discussed let's only check ~10 ids and put them all on one/two lines for convenience